### PR TITLE
RDM-2885: Delete Draft Definition

### PR DESCRIPTION
**Pull request by Daniel-Lam**
### JIRA link ###
https://tools.hmcts.net/jira/browse/RDM-2885.

### Change description ###
Initial version of Delete Draft Definition functionality.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/confirmDeleteItem.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/* GET */
+router.get("/deleteitem", (req, res, next) => {
+  const responseContent: { [k: string]: any } = {};
+  responseContent.response = req.session.response;
+  responseContent.itemToDelete = req.query.item;
+
+  if (req.query.item === "user") {
+    responseContent.idamId = req.query.idamId;
+    responseContent.warning = `Are you sure you would like to delete user ${req.query.idamId}?`;
```
Comment (adr1ancho) [inline src/main/routes/confirmDeleteItem.ts]: Just an observation, many places use a double quote for a string and this line uses ``.

File: src/main/service/delete-definition-service.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import * as request from "superagent";
+import * as config from "config";
+import { Logger } from "@hmcts/nodejs-logging";
+
+export function deleteDefinition(req) {
+  const logger = Logger.getLogger(__filename);
+  // Note: Not a typo - the base URLs for creating and deleting a Definition are exactly the same, hence reusing the
+  // create one
+  const url = config.get("adminWeb.createdefinition_url");
```
Comment (adr1ancho) [inline src/main/service/delete-definition-service.ts]: Thinking maybe we shall introduce another url in config.  That we you can save your L7..L8

File: src/main/routes/confirmDeleteItem.ts (hunk 1 above)
Comment (Daniel-Lam) [reply src/main/routes/confirmDeleteItem.ts]: That's deliberate, given that the ``` is used for template strings, but this one contains no variables and is a regular string.

File: src/main/service/delete-definition-service.ts (hunk 2 above)
Comment (Daniel-Lam) [reply src/main/service/delete-definition-service.ts]: OK, I'll separate into its own config variable

File: src/test/routes/deleteDefinition.spec.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                    expect(res.statusCode).to.equal(302);
+                    expect(res.headers.location.startsWith("/definitions")).to.be.true;
+                });
+        });
+
+        it("should redirect to the Definitions list when Yes is chosen but an error occurred", () => {
+            idamServiceMock.resolveRetrieveUserFor("1", "admin");
+            idamServiceMock.resolveRetrieveServiceToken();
+
+            mock("http://localhost:4451")
```
Comment (adr1ancho) [inline src/test/routes/deleteDefinition.spec.ts]: A few places you call nock as nock and this one is mock.  Which one we shall use to make the whole damn thing consistent

Comment (Daniel-Lam) [reply src/test/routes/deleteDefinition.spec.ts]: Good point - I'll rename to whichever is used more widely :)

File: src/main/routes/confirmDeleteItem.ts (hunk 1 above)
Comment (adr1ancho) [reply src/main/routes/confirmDeleteItem.ts]: Thank you for reminding this old man

Comment (Daniel-Lam) [reply src/main/routes/confirmDeleteItem.ts]: :) :+1:

File: src/test/routes/deleteDefinition.spec.ts (hunk 3 above)
Comment (Daniel-Lam) [reply src/test/routes/deleteDefinition.spec.ts]: There does seem to be some consistency... for whatever reason, `nock` is imported as "mock" in all the `routes` tests, but as "nock" in all the `service` tests. Therefore, I'm going to leave this one as is.

**Pull request merged**